crates/bevy_reflect/src/serde/mod.rs

```diff
@@ -181,4 +185,197 @@ mod tests {
 
         assert!(expected.reflect_partial_eq(&result).unwrap());
     }
+
+    mod type_data {
+        use super::*;
+        use crate::from_reflect::FromReflect;
+        use crate::serde::serialize_reflect::{ReflectSerializeReflect, SerializeReflect};
+        use crate::serde::{DeserializeReflect, ReflectDeserializeReflect};
+        use crate::{ReflectFromReflect, TypePath};
+        use bevy_reflect_derive::reflect_trait;
+        use core::fmt::{Debug, Formatter};
+        use serde::de::{SeqAccess, Visitor};
+        use serde::ser::SerializeSeq;
+        use serde::{Deserializer, Serializer};
+        use std::sync::Arc;
+
+        #[reflect_trait]
```

Comment:
This entire test below is _EXACTLY_ what's needed by 🎆 Hanabi to remove the clunky usage of `typetag` and finally integrate properly with Bevy's reflection and scene serialization. This is a gold mine of information there. Thank you so much @MrGVSV for this! 🥰